airflow/api/common/trigger_dag.py

```diff
@@ -54,7 +54,8 @@ def _trigger_dag(
     if dag is None or dag_id not in dag_bag.dags:
         raise DagNotFound(f"Dag id {dag_id} not found")
 
-    execution_date = execution_date if execution_date else timezone.utcnow()
+    if execution_date is None:
+        execution_date = timezone.utcnow()
```

Comment:
Would `execution_date = execution_date or timezone.utcnow()` work?

Reply:
Yes. However, while other cases are `c = a or b` and we need to attribute something to `c`, this is `a = a or b`, so if `a` is trueish, it is not modified.

Reply:
Yeah and I don’t think that’s a problem for this particular case.